test/unit/assorted/server_selection_latency_window_utils.ts

```diff
@@ -0,0 +1,150 @@
+import { EJSON } from 'bson';
+import { expect } from 'chai';
+import { readdirSync, readFileSync } from 'fs';
+import { join } from 'path';
+import { promisify } from 'util';
+
+import { ReadPreference } from '../../../src/read_preference';
+import { ServerType, STATE_CONNECTED, TopologyType } from '../../../src/sdam/common';
+import { Server } from '../../../src/sdam/server';
+import { Topology } from '../../../src/sdam/topology';
+import { serverDescriptionFromDefinition } from './server_selection_spec_helper';
+
+interface ServerSelectionLatencyWindowTest {
+  description: string;
+  topology_description: {
+    type: TopologyType;
+    servers: ReadonlyArray<{ address: string; avg_rtt_ms: number; type: ServerType }>;
+  };
+
+  mocked_topology_state: ReadonlyArray<{
+    address: string;
+    operation_count: number;
+  }>;
+
+  iterations: number;
+
+  outcome: {
+    tolerance: number;
+    expected_frequencies: {
+      [key: string]: number;
+    };
+  };
+}
+
+type Outcome = ServerSelectionLatencyWindowTest['outcome'];
+type FrequencyMap = Outcome['expected_frequencies'];
+
+export function loadLatencyWindowTests(directory: string) {
+  const files = readdirSync(directory).filter(fileName => fileName.endsWith('.json'));
+
+  const tests: ServerSelectionLatencyWindowTest[] = [];
+
+  for (const fileName of files) {
+    const path = join(directory, fileName);
+    const contents = readFileSync(path, { encoding: 'utf-8' });
+    tests.push(EJSON.parse(contents) as ServerSelectionLatencyWindowTest);
+  }
+
+  return tests;
+}
+
+function compareResultsToExpected(
+  { tolerance, expected_frequencies }: Outcome,
+  observed_frequencies: FrequencyMap
```

Comment:
can we consistently use camelCase for variables that we define vs props we read from the input file? it's rather disorienting to read the mix and match. 
1. For example, the variable in this function is defined as `observed_frequencies`, but the function itself is called with `observedFrequencies`. 
2. I see we already convert the `expected_frequencies` to `expectedFrequencies` below in L56, but then reference it directly again in L61 instead of using the variable we defined.
3. actual_frequency, is_too_high, and is_too_low aren't read from the input, so there doesn't really seem to be a reason to use underscores there